lib/elixir/lib/keyword.ex

```diff
@@ -650,18 +686,28 @@ defmodule Keyword do
   """
   @spec merge(t, t) :: t
   def merge(keywords1, keywords2) when is_list(keywords1) and is_list(keywords2) do
-    if keyword?(keywords2) do
-      fun = fn
-        {key, _value} when is_atom(key) ->
-          not has_key?(keywords2, key)
-        _ ->
-          raise ArgumentError, message: "expected a keyword list as the first argument, got: #{inspect keywords1}"
+    set =
+      try do
+        :maps.from_list(keywords2)
+      catch
+        :error, :badarg ->
+          raise ArgumentError, "expected a keyword list as the second argument, got: #{inspect keywords2}"
```

Comment:
The change here is that now we don't validate the second keyword fully, `[{"a", 2}]` would be accepted while it wasn't before.